### PR TITLE
Rename folders

### PR DESCRIPTION
**Pull request by miguelgfierro**
### Related Issues

https://github.com/microsoft/recommenders/issues/1390

### Checklist:

- [ ] I have followed the [contribution guidelines](../CONTRIBUTING.md) and code style for this project.
- [ ] I have added tests covering my contributions.
- [ ] I have updated the documentation accordingly.
- [ ] This PR is being made to `staging branch` and not to `main branch`.

### CODE REVIEW

**Comment by miguelgfierro** [comment]
@anargyri I'm able to render the formulas in reviewnb, however, this one is not rendering:

```
\textrm{IL} = \frac{1}{|M|} \sum_{u \in M} \frac{1}{\binom{N_r(u)}{2}} \sum_{i,j \in N_r (u),\, i
```
at first sight there is a } missing

File: tests/unit/examples/test_notebooks_python.py [hunk 1]
```diff
@@ -80,6 +81,7 @@ def test_lightgbm(notebooks, output_notebook, kernel_name):
 
 
 @pytest.mark.notebooks
+@pytest.mark.skip(reason="Wikidata API is unstable")
 def test_wikidata_runs(notebooks, output_notebook, kernel_name, tmp):
     notebook_path = notebooks["wikidata_knowledge_graph"]
     MOVIELENS_SAMPLE_SIZE = 5
```
Comment (miguelgfierro) [inline tests/unit/examples/test_notebooks_python.py:87]: @anargyri this is how tests can be skipped

File: examples/00_quick_start/sar_movielens.ipynb [hunk 2]
```diff
@@ -497,7 +497,7 @@
    "source": [
     "### 2.3. Evaluate how well SAR performs\n",
     "\n",
-    "We evaluate how well SAR performs for a few common ranking metrics provided in the `python_evaluation` module in reco_utils. We will consider the Mean Average Precision (MAP), Normalized Discounted Cumalative Gain (NDCG), Precision, and Recall for the top-k items per user we computed with SAR. User, item and rating column names are specified in each evaluation method."
```
Comment (anargyri) [inline examples/00_quick_start/sar_movielens.ipynb:500]: Revert this.

File: examples/02_model_collaborative_filtering/cornac_bpr_deep_dive.ipynb [hunk 3]
```diff
@@ -89,7 +89,7 @@
     "\n",
     "### 1.1 Personalized Ranking from Implicit Feedback\n",
     "\n",
-    "The task of personalized ranking aims at providing each user a ranked list of items (recommendations).  This is very common in scenarios where recommender systems are based on implicit user behavior (e.g. purchases, clicks).  The available observations are only positive feedback where the non-observed ones are a mixture of real negative feedback and missing values.\n",
```
Comment (anargyri) [inline examples/02_model_collaborative_filtering/cornac_bpr_deep_dive.ipynb:92]: This one too

File: examples/02_model_collaborative_filtering/cornac_bpr_deep_dive.ipynb [hunk 4]
```diff
@@ -118,7 +118,7 @@
     "\n",
     "The preference scoring function $\\hat{x}_{uij}(\\Theta)$ could be an arbitrary real-valued function of the model parameter $\\Theta$.  Thus, it makes BPR a general framework for modeling the relationship between triplets $(u, i, j)$ where different model classes like matrix factorization could be used for estimating $\\hat{x}_{uij}(\\Theta)$.\n",
     "\n",
-    "For the prior, one of the common pratices is to choose $p(\\Theta)$ following a normal distribution, which results in a nice form of L2 regularization in the final log-form of the objective function.\n",
```
Comment (anargyri) [inline examples/02_model_collaborative_filtering/cornac_bpr_deep_dive.ipynb:121]: Here too

File: examples/02_model_collaborative_filtering/lightgcn_deep_dive.ipynb [hunk 5]
```diff
@@ -728,7 +728,7 @@
     "\n",
     "Here there are the performances of LightGCN compared to [SAR](../00_quick_start/sar_movielens.ipynb) and [NCF](../00_quick_start/ncf_movielens.ipynb) on MovieLens dataset of 100k and 1m. The method of data loading and splitting is the same as that described above and the GPU used was a GeForce GTX 1080Ti.\n",
     "\n",
-    "Settings common to the three models: `epochs=15, seed=42`.\n",
```
Comment (anargyri) [inline examples/02_model_collaborative_filtering/lightgcn_deep_dive.ipynb:731]: +1

File: examples/02_model_hybrid/ncf_deep_dive.ipynb [hunk 6]
```diff
@@ -196,7 +196,7 @@
     "\n",
     "To evaluate the performance of item recommendation, we adopted the leave-one-out evaluation.\n",
     "\n",
-    "For each user, we held out his/her latest interaction as the test set and utilized the remaining data for training. We use `python_chrono_split` to achieve this. And since it is too time-consuming to rank all items for every user during evaluation, we followed the common strategy that randomly samples 100 items that are not interacted by the user, ranking the test item among the 100 items. Our test samples will be constructed by `NCFDataset`."
```
Comment (anargyri) [inline examples/02_model_hybrid/ncf_deep_dive.ipynb:199]: +1

File: examples/05_operationalize/als_movie_o16n.ipynb [hunk 7]
```diff
@@ -78,7 +78,7 @@
     
     "import urllib\n",
     "\n",
-    "from azure.common.client_factory import get_client_from_cli_profile\n",
```
Comment (anargyri) [inline examples/05_operationalize/als_movie_o16n.ipynb:81]: +1

File: examples/05_operationalize/als_movie_o16n.ipynb [hunk 8]
```diff
@@ -165,7 +165,7 @@
     "1. [Azure ML Service](https://azure.microsoft.com/en-us/services/machine-learning-service/)\n",
     "    1. [Azure ML Workspace](https://docs.microsoft.com/en-us/azure/machine-learning/concept-workspace)\n",
     "    1. [Azure Application Insights](https://azure.microsoft.com/en-us/services/monitor/)\n",
-    "    1. [Azure Storage](https://docs.microsoft.com/en-us/azure/storage/common/storage-account-overview)\n",
```
Comment (anargyri) [inline examples/05_operationalize/als_movie_o16n.ipynb:168]: +1

File: examples/00_quick_start/sar_movielens.ipynb (hunk 2 above)
Comment (miguelgfierro) [reply examples/00_quick_start/sar_movielens.ipynb:500]: wow, this is so weird, don't understand why those lines were deleted

File: reco_utils/evaluation/python_evaluation.py [hunk 9]
```diff
@@ -352,7 +352,7 @@ def merge_ranking_true_pred(
     k=DEFAULT_K,
     threshold=DEFAULT_THRESHOLD,
 ):
-    """Filter truth and prediction data frames on common users
```
Comment (anargyri) [inline reco_utils/evaluation/python_evaluation.py:355]: +1

**Comment by wutaomsft** [comment]
This work will trigger a new major version, correct? @miguelgfierro

**Comment by anargyri** [comment]
Wrong renames:

 reco_utils/dataset/__init__.py → reco_utils/models/__init__.py 

 reco_utils/recommender/__init__.py → reco_utils/models/cornac/__init__.py

File: reco_utils/evaluation/python_evaluation.py (hunk 9 above)
Comment (miguelgfierro) [reply reco_utils/evaluation/python_evaluation.py:355]: fixed

File: examples/00_quick_start/sar_movielens.ipynb (hunk 2 above)
Comment (anargyri) [reply examples/00_quick_start/sar_movielens.ipynb:500]: They were not deleted, you just changed a single word.

File: examples/05_operationalize/als_movie_o16n.ipynb (hunk 7 above)
Comment (miguelgfierro) [reply examples/05_operationalize/als_movie_o16n.ipynb:81]: fixed

File: examples/02_model_collaborative_filtering/lightgcn_deep_dive.ipynb (hunk 5 above)
Comment (miguelgfierro) [reply examples/02_model_collaborative_filtering/lightgcn_deep_dive.ipynb:731]: fixed

File: examples/05_operationalize/als_movie_o16n.ipynb (hunk 8 above)
Comment (miguelgfierro) [reply examples/05_operationalize/als_movie_o16n.ipynb:168]: fixed

File: examples/02_model_hybrid/ncf_deep_dive.ipynb (hunk 6 above)
Comment (miguelgfierro) [reply examples/02_model_hybrid/ncf_deep_dive.ipynb:199]: fixed

File: examples/02_model_collaborative_filtering/cornac_bpr_deep_dive.ipynb (hunk 4 above)
Comment (miguelgfierro) [reply examples/02_model_collaborative_filtering/cornac_bpr_deep_dive.ipynb:121]: fixed

File: examples/02_model_collaborative_filtering/cornac_bpr_deep_dive.ipynb (hunk 3 above)
Comment (miguelgfierro) [reply examples/02_model_collaborative_filtering/cornac_bpr_deep_dive.ipynb:92]: fixed

**Comment by miguelgfierro** [comment]
about this @anargyri:

>Wrong renames:
> reco_utils/dataset/__init__.py → reco_utils/models/__init__.py 
> reco_utils/recommender/__init__.py → reco_utils/models/cornac/__init__.py 
>reco_utils/recommender/cornac/__init__.py → ...tils/models/deeprec/DataModel/__init__.py 
>...recommender/deeprec/DataModel/__init__.py → reco_utils/models/deeprec/__init__.py 
> reco_utils/recommender/deeprec/__init__.py → reco_utils/models/deeprec/io/__init__.py 

I think the changes are ok, when I did git mv, I think git can't assign between init files because they are the exact same empty file

**Comment by miguelgfierro** [comment]
> This work will trigger a new major version, correct? @miguelgfierro

@wutaomsft we could do 0.6.1 if people want. Actually, now that we are changing paths, maybe we should consider other changes:
1. Do we want to change reco_utils to recommenders?
2. Do we want to change the folder of hybrid and move it to either CBF or CF?

I'll start a chat on teams about this

**Review by anargyri — APPROVED** [review]

**Pull request merged**